lib/getWorkflow.js

```diff
@@ -86,16 +86,24 @@ const calculateNodes = async (jobs, triggerFactory, externalDownstreamOrs, exter
         );
     }
 
+    const annotations = [
+        { annotation: 'annotations>screwdriver.cd/displayName', fieldName: 'displayName', default: undefined },
+        { annotation: 'annotations>screwdriver.cd/virtualJob', fieldName: 'virtual', default: undefined }
+    ];
+
     return [...nodes].map(name => {
         const m = { name };
-        const displayName = hoek.reach(jobs[name], 'annotations>screwdriver.cd/displayName', {
-            separator: '>',
-            default: undefined
-        });
 
-        if (displayName !== undefined) {
-            m.displayName = displayName;
-        }
+        annotations.forEach(({ annotation, fieldName, defaultValue }) => {
+            const fieldValue = hoek.reach(jobs[name], annotation, {
+                separator: '>',
+                default: defaultValue
+            });
+
+            if (fieldValue !== undefined) {
+                m[fieldName] = fieldValue;
```

Comment:
small suggestion: is using `value` a little redundant for variable names?

```suggestion
        annotations.forEach(({ annotation, fieldName, default }) => {
            const field = hoek.reach(jobs[name], annotation, {
                separator: '>',
                default
            });

            if (field !== undefined) {
                m[fieldName] = field;
```

Reply:
Also `default` is a keyword and cannot be used as variable name.